addons/l10n_hu_edi/wizard/l10n_hu_edi_tax_audit_export.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models, fields, _
+from odoo.exceptions import UserError
+
+import base64
+import tempfile
+import zipfile
+import contextlib
+
+class L10nHuEdiTaxAuditExport(models.TransientModel):
+    _name = 'l10n_hu_edi.tax_audit_export'
+    _description = 'Tax Audit Export Wizard'
+
+    date_from = fields.Date(
+        string='Date From'
+    )
+    date_to = fields.Date(
+        string='Date To'
+    )
+    filename = fields.Char(
+        string='File name',
+        compute='_compute_filename'
+    )
+    export_file = fields.Binary(
+        string='Generated File',
+        readonly=True
+    )
+
+    @api.depends('date_from', 'date_to')
+    def _compute_filename(self):
+        date_from = self.date_from
+        if not date_from:
+            first_invoice = self.env['account.move'].search(
+                [
+                    ('move_type', 'in', ('out_invoice', 'out_refund')),
+                    ('state', '=', 'posted'),
+                    ('country_code', '=', 'HU'),
+                ],
+                order='date',
+                limit=1,
+            )
+            date_from = first_invoice.date
+        date_to = self.date_to or fields.Date.today()
+        self.filename = f'export_{date_from}_{date_to}.zip'
+
+    def action_export(self):
+        self.ensure_one()
+        domain = [
+            ('move_type', 'in', ('out_invoice', 'out_refund')),
+            ('state', '=', 'posted'),
+            ('country_code', '=', 'HU'),
+        ]
+        if self.date_from:
+            domain.append(('date', '>=', self.date_from))
+        if self.date_to:
+            domain.append(('date', '<=', self.date_to))
+
+        invoices = self.env['account.move'].search(domain)
+        if not invoices:
+            raise UserError(_('No invoice to export!'))
+
+        with tempfile.NamedTemporaryFile(mode='w+b', delete=True) as buf:
```

Comment:
Thanks. I did this because I saw it done this way for the SBR export in the Netherlands. However, that was probably due for memory overflow reasons, and it might not be needed here.